### PR TITLE
Logs endpoint fixes

### PR DESCRIPTION
**Pull request by twometresteve**
## Goal

Corrects some issues with the new `/logs` endpoint:
- Disables `Bugsnag-Integrity` header checking for all requests on the endpoint
- Fails gracefully if log message checks are attempts before waiting to receive them
- Prints logs received at the end of any failing scenarios

## Tests

Testing using the in-development Android test fixture on branch `PLAT-5905/ndk-on-error`.  The scenarios did not pass as-is (I assume because it is still in development), but I was able to demonstrate the receipt and checking of a log message from a real test fixture.

### CODE REVIEW

**Review by Cawllec — APPROVED** [review]

File: lib/maze/server.rb
Comment (fractalwrench) [inline lib/maze/server.rb:126]: Should Bugsnag-Integrity checks be enabled for requests made to the /build endpoint?

**Review by fractalwrench — APPROVED** [review]
LGTM pending one query

File: lib/maze/server.rb
Comment (twometresteve) [reply lib/maze/server.rb:126]: There are now a few levels of control now as to when the integrity header will be enforced.  This is the most blunt level of control before it flows down to the MR servlet (which will not enforce it for `multipart/form-data` requests).  Whilst I wouldn't currently expect any build requests to have the integrity header, I think it's reasonable to leave it to the downstream mechanisms to control because, in theory, the product change change and include the digest.  Log requests, however, are not a part of the product and can therefore be blanket excluded.

**Pull request merged**
